File: examenes-resueltos/2024-junio-harleydawson/backend/model/Delivery.java
@Entity
public class Delivery {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String plate;

    private String model;

    private String color;

    private int displacement;

    private String address;

    private String clientName;

    private String status; // PENDING, CANCELLED, FINALIZED

    public Delivery() {
    }

    public Delivery(String plate, String model, String color,
                    int displacement, String address,
                    String clientName, String status) {
        this.plate = plate;
        this.model = model;
        this.color = color;
        this.displacement = displacement;
        this.address = address;
        this.clientName = clientName;
        this.status = status;
    }

    // getters and setters
}

File: examenes-resueltos/2024-junio-harleydawson/frontend/models/Delivery.ts
// Interfaz que representa una entrega de moto
export default interface Delivery {
    id?: number;
    plate: string;
    model: string;
    color: string;
    displacement: number;
    address: string;
    clientName: string;
    status: string; // PENDING, CANCELLED, FINALIZED
}
